lib/ansible/modules/network/iosxr/iosxr_l3_interfaces.py

```diff
@@ -0,0 +1,411 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 Red Hat Inc.
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+##############################################
+#                 WARNING
+#
+# This file is auto generated by the resource
+#   module builder playbook.
+#
+# Do not edit this file manually.
+#
+# Changes to this file will be over written
+#   by the resource module builder.
+#
+# Changes should be made in the model used to
+#   generate this file or in the resource module
+#   builder template.
+#
+#
+##############################################
+
+"""
+The module file for ios_l3_interfaces
+"""
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'network'}
+
+
+DOCUMENTATION = """
+module: iosxr_l3_interfaces
+  version_added: 2.9
+  short_description: Manage Layer-3 interface on Cisco IOS-XR devices.
+  description: This module provides declarative management of Layer-3 interface on Cisco IOS-XR devices.
+  author: Sumit Jaiswal (@justjais)
```

Comment:
We should add the `notes` section specifying the IOS-XR version tested against and the type of connection this module supports.